pkg/tfgen/installation_docs.go

```diff
@@ -245,3 +250,111 @@ func convertExample(g *Generator, code string, exampleNumber int) (string, error
 	exampleContent += chooserEnd
 	return exampleContent, nil
 }
+
+type sectionSkipper struct {
+	shouldSkipHeader func(headerText string) bool
+}
+
+var _ parser.ASTTransformer = sectionSkipper{}
+
+func (t sectionSkipper) Transform(node *ast.Document, reader text.Reader, pc parser.Context) {
+	source := reader.Source()
+	removingLevel := 0
+	// All headings are first children of the ast.Document node.
+	// We will walk over them and remove any that match the header content we do not want.
+	// Additionally, we will track the header level and remove any content that is nested under the removed header.
+	for currentChild := node.FirstChild(); currentChild != nil; {
+		if removingLevel > 0 {
+			if child, ok := currentChild.(*ast.Heading); ok && child.Level <= removingLevel {
+				removingLevel = 0
+			} else {
+				nextNode := currentChild.NextSibling()
+				currentChild.Parent().RemoveChild(currentChild.Parent(), currentChild)
+				currentChild = nextNode
+				continue
+			}
+		}
+		// In case our level got reset, we check again
+		if removingLevel == 0 {
+			switch child := currentChild.(type) {
+			case *ast.Heading:
+				header := child.Text(source)
+				if t.shouldSkipHeader(string(header)) {
+					removingLevel = child.Level
+					nextNode := child.NextSibling()
+					child.Parent().RemoveChild(child.Parent(), child)
+					currentChild = nextNode
+					continue
+				}
+			}
+		}
+		// Move to next node in base case.
+		currentChild = currentChild.NextSibling()
+	}
+}
+
+// SkipSectionByHeaderContent removes headers where shouldSkipHeader(header) returns true,
+// along with any text under the header.
+// content is assumed to be Github flavored markdown when parsing.
+//
+// shouldSkipHeader is called on the raw header text, like this:
+//
+//	shouldSkipHeader("My Header")
+//
+// *not* like this:
+//
+//	// This is wrong
+//	shouldSkipHeader("## My Header\n")
```

Comment:
Perhaps a full example of calling this function would help demonstrate how to write the callback as well as indicate how the header is matched ...

```suggestion
// SkipSectionByHeaderContent removes headers where shouldSkipHeader(header) returns true,
// along with any text under the header.
// content is assumed to be Github flavored markdown when parsing.
//
// shouldSkipHeader is called on the parsed header text.
// 
// Example of removing the first header and it's context
//
//	result := SkipSectionByHeaderContent(byte[](`
//	## First Header
//	
//	First content
//	
//	## Second Header
//	
//	Second content
//	`, func(headerText string) bool) ([]byte, error) {
//		return headerText == "First Header"
//	})
// 
// The result will only now contain the second result:
//	## Second Header
//	
//	Second content
```